ipam/allocator.go

```diff
@@ -473,8 +473,11 @@ func (alloc *Allocator) encode() []byte {
 	}
 
 	// We're only interested in Paxos until we have a Ring.
+	// Non-electing participants (e.g. observers) return
+	// a nil gossip state in order to provoke a unicast ring
+	// update from electing peers who have reached consensus.
```

Comment:
I don't think that is sufficient.

When an ip allocation happens on a non-elector, this should prompt the establishment of a ring among the electors, even though none of them themselves may have required an election.

Otherwise IP allocation on a non-elector could be stalled, i.e. when non of the electors require allocation.